org.karapulse.Karapulse.json

```diff
@@ -0,0 +1,114 @@
+            "CARGO_HOME": "/run/build/Karapulse/cargo",
+            "RUST_BACKTRACE": "1",
+            "RUST_LOG": "karapulse=debug"
+        }
+    },
+    "modules": [
+        {
+            "name": "rust",
+            "buildsystem": "simple",
+            "sources": [
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "x86_64"
+                    ],
+                    "url": "https://static.rust-lang.org/dist/rust-nightly-x86_64-unknown-linux-gnu.tar.gz",
+                    "sha256": "8c714c2fc359fe55f121859dbefc5ac0f722c10b7e1ebf5657bf455c57be5376"
+                },
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "i386"
+                    ],
+                    "url": "https://static.rust-lang.org/dist/rust-nightly-i686-unknown-linux-gnu.tar.gz",
+                    "sha256": "c420eca932096ed1a48d75e6a2b37acdf1c0d4e4398ae7e8538c5f44addd0f2c"
+                },
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "arm"
+                    ],
+                    "url": "https://static.rust-lang.org/dist/rust-nightly-arm-unknown-linux-gnueabihf.tar.gz",
+                    "sha256": "745f9eebd814536ba32079405810c78442d8e3eb6ed1c922354f8ff90f6fd923"
+                },
+                {
+                    "type": "archive",
+                    "only-arches": [
+                        "aarch64"
+                    ],
+                    "url": "https://static.rust-lang.org/dist/rust-nightly-aarch64-unknown-linux-gnu.tar.gz",
+                    "sha256": "61e821b297703878d32b62a5459840e6db5e7e3f11c72135e37a24ebc335a8ae"
+                }
+            ],
+            "build-commands": [
+                "./install.sh --prefix=/app/sdk/rust --disable-ldconfig --verbose"
+            ]
+        },
+        {
+            "name": "nodejs",
+            "cleanup": [
+                "/include",
+                "/share",
+                "/app/lib/node_modules/npm/changelogs",
+                "/app/lib/node_modules/npm/doc",
+                "/app/lib/node_modules/npm/html",
+                "/app/lib/node_modules/npm/man",
+                "/app/lib/node_modules/npm/scripts"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://nodejs.org/dist/v8.9.2/node-v8.9.2.tar.gz",
+                    "sha256": "8038fa61ac4562786fcc6d3229f3caa07a864b94bd5922fe75942a38ea1c3a3b"
+                }
+            ]
+        },
+        {
+            "name": "karapulse",
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dcdg-plugin=true"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/gdesmott/karapulse.git"
```

Comment:
Same as for network access during the build; we want reproducible builds, so a tarball/tag/commit is needed.

Reply:
My plan was to point to a release git tag. But I don't understand how this would fit with the 'dist' approach we discussed.
Is it ok if 'ninja dist` download stuffs and create a tarball with all the deps? Is it ok to generate the tarball during the flatpak build?

Reply:
I assume that `ninja dist` puts required dependencies as part of the tarball? Then that would be the way to go, or the generator I linked above.